psycop/projects/scz_bp/text_experiment/mlflow_data_extraction.py

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Iterable
+
+import polars as pl
+from mlflow.entities import Run
+from mlflow.store.entities import PagedList
+from mlflow.tracking import MlflowClient
+
+
+class MlflowDataExtraction:
+    def __init__(self) -> None:
+        self.client = MlflowClient(tracking_uri="http://exrhel0371.it.rm.dk:5050")
+
+    def get_all_metrics_for_experiment(self, experiment_name: str) -> pl.DataFrame:
+        """Get the final value of all logged metrics for each run in an experiment.
+        Returns a long df with cols 'run_name', 'metric', and 'value'."""
+        runs = self._get_mlflow_runs_by_experiment(experiment_name=experiment_name)
+        metrics_df = [self._get_all_metrics_for_run(run) for run in runs]
+        return pl.concat(metrics_df).melt(id_vars="run_name", variable_name="metric")
+
+    def get_metrics_for_experiment(
+        self, experiment_name: str, metrics: Iterable[str]
+    ) -> pl.DataFrame:
+        """Get all values of specified metrics for each run in an experiment. This
+        includes values that get updated such as loss. Returns a long df with
+        cols 'run_name', 'metric', 'value', 'timestamp', 'step'."""
+        runs = self._get_mlflow_runs_by_experiment(experiment_name=experiment_name)
+        metrics_df = [self._get_metrics_for_run(run=run, metrics=metrics) for run in runs]
+        return pl.concat(metrics_df).rename({"key": "metric"})
+
+    def _get_metrics_for_run(self, run: Run, metrics: Iterable[str]) -> pl.DataFrame:
+        metrics_df = [
+            self._get_metric_for_run(run_id=run.info.run_id, metric=metric) for metric in metrics
+        ]
+        return pl.concat(metrics_df).with_columns(pl.lit(run.info.run_name).alias("run_name"))
+
+    def _get_metric_for_run(self, run_id: str, metric: str) -> pl.DataFrame:
+        return pl.DataFrame(
+            [
+                dict(metric_object)
+                for metric_object in self.client.get_metric_history(run_id=run_id, key=metric)
+            ]
```

Comment:
Just want to highlight an Iterpy solution here:

`return Iter(self.client.get_metric_history(run_id=run_id, key=metric)).map(dict).to_list()`

Reply:
Huh, I expected that to be much more readable, but it was only OK 🤷‍♂️

Reply:
Hm, yeah - not sure which I prefer tbh 🤷‍♂️